plugins/callback/opentelemetry.py

```diff
@@ -110,13 +110,31 @@
     from opentelemetry.sdk.trace.export import (
         BatchSpanProcessor
     )
-    from opentelemetry.util._time import _time_ns
+
+    # Support for opentelemetry-api <= 1.12
+    try:
+        from opentelemetry.util._time import _time_ns
+    except ImportError as imp_exc:
+        OTEL_LIBRARY_TIME_NS_ERROR = imp_exc
+    else:
+        OTEL_LIBRARY_TIME_NS_ERROR = None
+
 except ImportError as imp_exc:
     OTEL_LIBRARY_IMPORT_ERROR = imp_exc
 else:
     OTEL_LIBRARY_IMPORT_ERROR = None
 
 
+def time_ns():
+    if sys.version_info >= (3, 7):
+        return time.time_ns()
+    else:
+        # Support versions older than 3.7 with opentelemetry-api > 1.12
+        if OTEL_LIBRARY_TIME_NS_ERROR:
+            return int(time.time() * 1e9)
+        return _time_ns()
```

Comment:
Given we are talking about that level of precision in time, we might want to remove that branching from  inside the function:
```suggestion
if sys.version_info >= (3, 7):
    def time_ns():
        return time.time_ns()
elif not OTEL_LIBRARY_TIME_NS_ERROR:
    def time_ns():
        return _time_ns()
else:
    def time_ns():
        # Support versions older than 3.7 with opentelemetry-api > 1.12
        return int(time.time() * 1e9)
```

Reply:
Why not `time_ns = time.time_ns` instead of
```
def time_ns():
    return time.time_ns()
```
?

Reply:
Indeed

Reply:
I'm more than happy to commit the suggestion, please bear with me, but what's the benefit from using the branch outside of the function when the function scope is well defined? Is that the standard way of doing it, just I know this for the future, 🙏 .

Reply:
Hi @v1v the idea is that, with the branch inside, those conditions will have to be tested every time the function is called.

Given that those conditions are somewhat static, more related to the environment and unlikely to change during the module runtime, we can make the decision earlier and have a branch-less function, effectively meaning that any of the three functions will be faster than the original one.

Normally I wouldn't bother with this as the difference in performance is quite small. The reason why I am suggesting this here is exactly because we're dealing with measuring nanoseconds, so it is more likely to be a noticeable difference in this context.

Reply:
@v1v basically performance / speed. When the `if` is outside the function definition, determining which function to use when `time_ns` is called is done once (on plugin loading). When the `if` is inside the function definition, determining how to get the time is done every time `time_ns` is called.

The practical implications of this speed improvements are probably not that relevant, but since it also doesn't really decrease readability, there's no "obvious" answer to whether to apply this or not.

Reply:
(Meh, I only noticed after writing and sending this that @russoz already commented on this...)

Reply:
Thanks for the answers and help me to understand your proposal 🙏